infrastructure/src/main/java/org/corfudb/infrastructure/CompactorLeaderServices.java

```diff
@@ -240,6 +244,14 @@ public void finishCompactionCycle() {
                     tableNames.size(), finalStatus);
             MicroMeterUtils.time(Duration.ofMillis(totalTimeElapsed), "compaction.total.timer",
                     "nodeEndpoint", nodeEndpoint);
+            Issue compactionCycleIssue =
+                    Issue.createIssue(COMPACTOR, COMPACTION_CYCLE_FAILED, "Last compaction cycle failed");
+
+            if (finalStatus == StatusType.FAILED) {
+                HealthMonitor.reportIssue(compactionCycleIssue);
+            } else {
+                HealthMonitor.resolveIssue(compactionCycleIssue);
+            }
```

Comment:
Do we want to report/resolve the issue for other code paths in this method as well? This could be moved to a finally clause so that it can still run in the case of an exception.

Reply:
Good call, fixed